server/src/main/scala/za/co/absa/atum/server/api/repository/FlowRepositoryImpl.scala

```diff
@@ -16,25 +16,48 @@
 
 package za.co.absa.atum.server.api.repository
 
-import za.co.absa.atum.model.dto.CheckpointQueryDTO
-import za.co.absa.atum.server.api.database.flows.functions.GetFlowCheckpoints
+import za.co.absa.atum.model.dto.CheckpointV2DTO
+import za.co.absa.atum.server.api.database.flows.functions.GetFlowCheckpointsV2
 import za.co.absa.atum.server.api.exception.DatabaseError
-import za.co.absa.atum.server.model.CheckpointFromDB
+import za.co.absa.atum.server.model.{CheckpointItemFromDB, PaginatedResult}
+import za.co.absa.atum.server.api.database.flows.functions.GetFlowCheckpointsV2.GetFlowCheckpointsArgs
+import za.co.absa.atum.server.api.exception.DatabaseError.GeneralDatabaseError
+import za.co.absa.atum.server.model.PaginatedResult.{ResultHasMore, ResultNoMore}
 import zio._
 import zio.interop.catz.asyncInstance
 
-class FlowRepositoryImpl(getFlowCheckpointsFn: GetFlowCheckpoints) extends FlowRepository with BaseRepository {
+class FlowRepositoryImpl(getFlowCheckpointsV2Fn: GetFlowCheckpointsV2)
+  extends FlowRepository with BaseRepository {
 
-  override def getFlowCheckpoints(checkpointQueryDTO: CheckpointQueryDTO): IO[DatabaseError, Seq[CheckpointFromDB]] = {
-    dbMultipleResultCallWithAggregatedStatus(getFlowCheckpointsFn(checkpointQueryDTO), "getFlowCheckpoints")
-  }
+  override def getFlowCheckpointsV2(
+    partitioningId: Long,
+    limit: Option[Int],
+    offset: Option[Long],
+    checkpointName: Option[String] = None,
+   ): IO[DatabaseError, PaginatedResult[CheckpointV2DTO]] = {
+      dbMultipleResultCallWithAggregatedStatus(
+        getFlowCheckpointsV2Fn(GetFlowCheckpointsArgs(partitioningId, limit, offset, checkpointName)),
+        "getPartitioningCheckpoints"
```

Comment:
do not blindly copy/paste stuff, this is not a getPartitioningCheckpoints operation